climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,891 @@
+
+    """
+
+    if poly.is_empty:
+        return shgeom.MultiPoint([])
+
+    repr_pnt = poly.representative_point()
+    lon_0, lat_0 = repr_pnt.x, repr_pnt.y
+
+    project = pyproj.Transformer.from_proj(
+        pyproj.Proj(orig_crs),
+        pyproj.Proj("+proj=cea +lat_0=%f +lon_0=%f +units=m" %(lat_0, lon_0)),
+        always_xy=True
+    )
+    poly_m = sh.ops.transform(project.transform, poly)
+
+    height, width, trafo = u_coord.pts_to_raster_meta(poly_m.bounds, (res, res))
+    x_grid, y_grid = u_coord.raster_to_meshgrid(trafo, width, height)
+
+    in_geom = sh.vectorized.contains(poly_m, x_grid, y_grid)
+
+    if sum(in_geom.flatten()) > 1:
+        project_inv = pyproj.Transformer.from_proj(
+            pyproj.Proj("+proj=cea +lat_0=%f +lon_0=%f +units=m" %(lat_0, lon_0)),
+            pyproj.Proj(orig_crs),
+            always_xy=True
+        )
+        x_poly, y_poly = project_inv.transform(x_grid[in_geom], y_grid[in_geom])
+        poly_pnt = shgeom.MultiPoint([(x, y) for x, y in
+                            zip(x_poly, y_poly)])
+    else:
+        poly_pnt = shgeom.MultiPoint([repr_pnt])
+
+    return poly_pnt
+
+
+def poly_to_equalarea_proj(poly, orig_crs):
+    """
+    Project polyong to Equal Area Cylindrical projection
+    using a representative point as lat/lon reference.
+
+    https://proj.org/operations/projections/cea.html
+
+    Parameters
+    ----------
+    poly : shapely Polygon
+        Polygon
+    orig_crs: pyproj.CRS
+        CRS of the polygon
+
+    Returns
+    -------
+    poly : shapely Polygon
+        Polygon in equal are projection
+
+    """
+
+    repr_pnt = poly.representative_point()
+    lon_0, lat_0 = repr_pnt.x, repr_pnt.y
+
+    project = pyproj.Transformer.from_proj(
+        pyproj.Proj(orig_crs),
+        pyproj.Proj("+proj=cea +lat_0=%f +lon_0=%f +units=m" %(lat_0, lon_0)),
+        always_xy=True
+    )
+    return sh.ops.transform(project.transform, poly)
+
+def line_to_pnts(gdf_lines, res):
+
+    """ Convert a GeoDataframe with LineString geometries to
+    Point geometries, where Points are placed at a specified distance
+    along the original LineString. Each line is reduced to
+    at least two points.
+
+    Parameters
+    ----------
+    gdf_lines : gpd.GeoDataframe
+        Geodataframe with line geometries
+    res : float
+        Resolution (distance) apart from which the generated Points
+        should be approximately placed.
+
+    Returns
+    -------
+    gdf_points : gpd.GeoDataFrame
+        Geodataframe with a double index, first for line geometries,
+        second for the point disaggregation of the lines (i.e. one Point
+        per row).
+
+    See also
+    --------
+    * util.coordinates.compute_geodesic_lengths()
+    """
+
+    # Needed because gdf.explode() requires numeric index
+    idx = gdf_lines.index.to_list() #To restore the naming of the index
+
+    gdf_points = gdf_lines.copy().reset_index(drop=True)
+    line_lengths = gdf_lines.length
```

Comment:
Does a gdf have a defined `.length` attribute by default?

Reply:
Yes. 

> In the case of a (Multi)Polygon it measures the length of its exterior (i.e. perimeter).

For Points it is 0.